modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -0,0 +1,1201 @@
+
+    $element['settings'] = [
+      '#type' => 'horizontal_tabs',
+      '#tree' => TRUE,
+      '#group_name' => 'settings',
+    ];
+    $element['settings']['content'] = [
+      '#type' => 'details',
+      '#open' => TRUE,
+      '#collapsible' => TRUE,
+      '#title' => $this->t('Content'),
+      '#group_name' => 'tabs_content',
+    ];
+
+    $element['settings']['content']['enable_call_to_action'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable call to action'),
+      '#default_value' => $settings['content']['enable_call_to_action'] ?? FALSE,
+      '#weight' => -1,
+    ];
+
+    if ($content_type_options = $this->indexHelper->getNodeTypes()) {
+      $element['settings']['content']['internal']['contentTypes'] = [
+        '#type' => 'details',
+        '#title' => 'Content Types',
+        '#open' => FALSE,
+        '#collapsible' => TRUE,
+        '#weight' => 2,
+      ];
+      $element['settings']['content']['internal']['contentTypes']['enabled'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Enable content types'),
+        '#default_value' => $settings['content']['internal']['contentTypes']['enabled'] ?? FALSE,
+      ];
+      $element['settings']['content']['internal']['contentTypes']['allowed_values'] = [
+        '#type' => 'checkboxes',
+        '#title' => $this->t('Allowed content types'),
+        '#description' => $this->t('When no content type is selected in the widget settings, the widget will show all available content types in the Select content type filter.'),
+        '#options' => $content_type_options,
+        '#default_value' => $settings['content']['internal']['contentTypes']['allowed_values'] ?? [],
+        '#weight' => 1,
+      ];
+      $element['settings']['content']['internal']['contentTypes']['default_values'] = [
+        '#type' => 'checkboxes',
+        '#title' => $this->t('Default content types'),
+        '#description' => $this->t('When no content type is selected in the widget settings, the widget will show all available content types in the Select content type filter.'),
+        '#options' => $content_type_options,
+        '#default_value' => $settings['content']['internal']['contentTypes']['default_values'] ?? [],
+        '#weight' => 1,
+        '#states' => [
+          'visible' => [
+            ':input[name="fields[' . $field_name . '][settings_edit_form][settings][settings][content][internal][contentTypes][enabled]"]' => ['checked' => FALSE],
+          ],
+        ],
+      ];
+    }
+
+    if ($this->indexHelper->isFieldTopicIndexed($index)) {
+      $element['settings']['content']['internal']['field_topic'] = [
+        '#type' => 'details',
+        '#title' => 'Field Topic',
+        '#open' => FALSE,
+        '#collapsible' => TRUE,
+        '#weight' => 2,
+      ];
+      $element['settings']['content']['internal']['field_topic']['enabled'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Enable field_topic'),
```

Comment:
Topic and Tags must have their name, label. etc. treated specially, eg. `Enable Topic filter`.